Templating/TemplateGuesser.php

```diff
@@ -30,22 +30,33 @@ class TemplateGuesser
     protected $kernel;
 
     /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel             A KernelInterface instance
+     * @param string[]        $controllerPatterns Regexps extracting the controller name from its FQN.
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, array $controllerPatterns = null)
     {
+        if (null === $controllerPatterns) {
+            $controllerPatterns = array('/Controller\\\(.+)Controller$/');
```

Comment:
As proposed by @dunglas https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/398#discussion-diff-51360644 the main benefit to implement it that way is that people can decide to not support their legacy controllers.
Both solutions have their own advantages so I don't know which one we should use...